tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/SignificantIndentationSuite.scala

```diff
@@ -0,0 +1,363 @@
+              Type.Name("T"),
+              Nil,
+              Ctor.Primary(Nil, Name(""), Nil),
+              Template(
+                Nil,
+                Nil,
+                Self(Name(""), None),
+                List(Defn.Def(Nil, Term.Name("f"), Nil, Nil, Some(Type.Name("Int")), Lit.Int(4)))
+              )
+            )
+          )
+        )
+      )
+    )
+  }
+
+  test("indent-below-not-okay") {
+    val code = """|def fn: Unit =
+                  |    if cond then
+                  |  truep
+                  |    else
+                  |  falsep
+                  |""".stripMargin
+    runTestError[Stat](code, "error: illegal start of simple expression")
+  }
+
+  test("indent-inside-brace-ok") {
+    val code = """|class X {
+                  |  def fx(): Unit =
+                  |    f1
+                  |    f2
+                  |}
+                  |""".stripMargin
+    val output = """|class X {
+                    |  def fx(): Unit = {
+                    |    f1
+                    |    f2
+                    |  }
+                    |}
+                    |""".stripMargin
+    runTestAssert[Stat](code, assertLayout = Some(output))(
+      Defn.Class(
+        Nil,
+        Type.Name("X"),
+        Nil,
+        Ctor.Primary(Nil, Name(""), Nil),
+        Template(
+          Nil,
+          Nil,
+          Self(Name(""), None),
+          List(
+            Defn.Def(
+              Nil,
+              Term.Name("fx"),
+              Nil,
+              List(List()),
+              Some(Type.Name("Unit")),
+              Term.Block(List(Term.Name("f1"), Term.Name("f2")))
+            )
+          )
+        )
+      )
+    )
+  }
+
+  test("indent-inside-brace-not") {
+    val code = """|class X {
+                  |      def fx(): Unit =
+                  |    f1
+                  |    f2
+                  |}
+                  |""".stripMargin
+    val output = """|class X {
+                    |  def fx(): Unit = f1
+                    |  f2
+                    |}
+                    |""".stripMargin
+    runTestAssert[Stat](code, assertLayout = Some(output))(
+      Defn.Class(
+        Nil,
+        Type.Name("X"),
+        Nil,
+        Ctor.Primary(Nil, Name(""), Nil),
+        Template(
+          Nil,
+          Nil,
+          Self(Name(""), None),
+          List(
+            Defn.Def(
+              Nil,
+              Term.Name("fx"),
+              Nil,
+              List(List()),
+              Some(Type.Name("Unit")),
+              Term.Name("f1")
+            ),
+            Term.Name("f2")
+          )
+        )
+      )
+    )
+  }
+
+  test("should-indent-yet-brace") {
+    val code = """|class X {
+                  |  def fx(): Unit =
+                  |  {
+                  |
+                  |  }
+                  |  private def f2: Int = 1
+                  |}
+                  |""".stripMargin
+    runTestAssert[Stat](code, assertLayout = None)(
```

Comment:
Maybe add assert layout? I find it useful to figure out the differences between dotty's significant indentation and previous normal approach

Reply:
added